src/privatesend-client.cpp

```diff
@@ -880,31 +884,23 @@ bool CPrivateSendClient::JoinExistingQueue(CAmount nBalanceNeedsAnonymized, CCon
             continue;
         }
 
-        LogPrintf("CPrivateSendClient::JoinExistingQueue -- attempt to connect to masternode from queue, addr=%s\n", infoMn.addr.ToString());
-        // connect to Masternode and submit the queue request
-        CAddress addr(infoMn.addr, NODE_NETWORK);
-        connman.OpenMasternodeConnection(addr);
-
-        bool fSuccess = connman.ForNode(addr, CConnman::AllNodes, [&](CNode* pnode){
-            infoMixingMasternode = infoMn;
-            nSessionDenom = dsq.nDenom;
-            CDarksendAccept dsa(nSessionDenom, txMyCollateral);
-            CNetMsgMaker msgMaker(pnode->GetSendVersion()); // TODO this gives a warning about version not being set (we should wait for VERSION exchange)
-            connman.PushMessage(pnode, msgMaker.Make(NetMsgType::DSACCEPT, dsa));
-            LogPrintf("CPrivateSendClient::JoinExistingQueue -- connected (from queue), sending DSACCEPT: nSessionDenom: %d (%s), addr=%s\n",
-                      nSessionDenom, CPrivateSend::GetDenominationsToString(nSessionDenom), pnode->addr.ToString());
-            strAutoDenomResult = _("Mixing in progress...");
-            SetState(POOL_STATE_QUEUE);
-            nTimeLastSuccessfulStep = GetTimeMillis();
-            return true;
-        });
-        if (!fSuccess) {
-            LogPrintf("CPrivateSendClient::JoinExistingQueue -- can't connect, addr=%s\n", infoMn.addr.ToString());
-            strAutoDenomResult = _("Error connecting to Masternode.");
-            continue;
+        nSessionDenom = dsq.nDenom;
+        infoMixingMasternode = infoMn;
+
+        {
+            LOCK(cs_dsaQueue);
+            dsaQueue.push_back(std::make_pair(GetTime(), std::make_pair(infoMn.addr, CDarksendAccept(nSessionDenom, txMyCollateral))));
         }
+
+        connman.AddPendingMasternode(infoMn.addr);
+        SetState(POOL_STATE_QUEUE);
+        nTimeLastSuccessfulStep = GetTimeMillis();
+        LogPrintf("CPrivateSendClient::JoinExistingQueue -- connecting (from queue): nSessionDenom: %d (%s), addr=%s\n",
+                  nSessionDenom, CPrivateSend::GetDenominationsToString(nSessionDenom), infoMn.addr.ToString());
+        strAutoDenomResult = _("Mixing in progress...");
         return true;
     }
+    strAutoDenomResult = _("Failed to join existing mixing queues");
```

Comment:
Would something like "Failed to find mixing queue to join" better describe what went wrong? (we didn't even try to join a queue as I understand it)